### PR TITLE
COMMERCE-8489, COMMERCE-8287 | master

### PR DESCRIPTION
**Pull request by Kvale1733**
https://issues.liferay.com/browse/COMMERCE-8489
https://issues.liferay.com/browse/COMMERCE-8287

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:commerce:nocompile

**Comment by Kvale1733** [comment]
ci:test:commerce:nocompile

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase [hunk 1]
```diff
@@ -2094,6 +2094,242 @@ definition {
...
+			FrontStore.addInitializerProductToCart(
+				productName = "U-Joint",
+				productQuantity = "1");
+
+			CommerceOrders.selectOrderFromAccountSelector(orderId = "${order2Id}");
+
+			Button.click(button = "Checkout");
+		}
+
+		task ("Then all active Shipping options should be available") {
```
Comment (iovaneale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase:2165]: @Kvale1733  Shouldn't we assert that all the expected shipping options are available? Here the macro executes the checkout and we pass the SO as variable so we're not sure whether the others are available or not. I'd suggest to add the logic to the macro by the use of something like:

if (isSet(assertAvailableShippingOptions)) {
AssertElementPresent [...]

with the var to have a value of true if we want to let the macro executing that block. If the solution fits the case, we should use it also in the following block.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase [hunk 2]
```diff
@@ -2094,6 +2094,242 @@ definition {
...
+			FrontStore.addInitializerProductToCart(
+				productName = "U-Joint",
+				productQuantity = "1");
+
+			CommerceOrders.selectOrderFromAccountSelector(orderId = "${order1Id}");
+
+			Button.click(button = "Checkout");
+		}
+
+		task ("Then all active Shipping options should be available") {
```
Comment (iovaneale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase:2194]: Same as above

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase [hunk 3]
```diff
@@ -2094,6 +2094,242 @@ definition {
...
+				newAddressAddress = "Test Address",
+				newAddressCity = "Test City",
+				newAddressCountry = "United States",
+				newAddressName = "Address Name",
+				newAddressZipCode = "Test Zip Code");
+
+			Click(locator1 = "Button#CONTINUE");
+		}
+
+		task ("Then the edited Shipping option should not be available") {
```
Comment (iovaneale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceChannels.testcase:2297]: Taking advantage of the comments above, if it doesn't affect the readability of the test, we could add in the if(isSet) block vars for each shipping option to set as true or false.

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:close

**Comment by iovaneale** [comment]
Sent to #2700.

**Pull request closed without merging**